blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/AbstractMetricsRW.java

```diff
@@ -148,22 +149,19 @@ protected int getTtl(Locator locator, RollupType rollupType, Granularity granula
      * @param timestampToRollupMap  a map of timestamp to rollup
      * @return
      */
-    protected <T extends Rollup> Points<T> convertToPoints(final Map<Long, T> timestampToRollupMap) {
-        Points<T> rollupPoints =  new Points<T>();
-        for (Map.Entry<Long, T> rollup : timestampToRollupMap.entrySet() ) {
-            rollupPoints.add(new Points.Point<T>(rollup.getKey(), rollup.getValue()));
+    protected Points convertToPoints(final Map<Long, Object> timestampToRollupMap) {
+        Points points =  new Points();
+        for (Map.Entry<Long, Object> value : timestampToRollupMap.entrySet() ) {
+
+            points.add( createPoint( value.getKey(), value.getValue() ) );
         }
-        return rollupPoints;
+        return points;
     }
 
-    /**
-     * Inserts a collection of rolled up metrics to the metrics_preaggregated_{granularity} column family.
-     * Only our tests should call this method. Services should call either insertMetrics(Collection metrics)
-     * or insertRollups()
-     *
-     * @param metrics
-     * @throws IOException
-     */
-    @VisibleForTesting
-    public abstract void insertMetrics(Collection<IMetric> metrics, Granularity granularity) throws IOException;
+    protected Points.Point createPoint( Long timestamp, Object value ) {
+        if( value instanceof Rollup || value instanceof String)
```

Comment:
I guess it's not necessary, it's just more explicit. It would also get rid of the instanceof. Why do you say for Basic metrics you'll have to do instanceof check? If we make a ```createPoint()``` method that takes ```String```, we still have to do instanceof String check?